www/base/src/app/settings/service/settings.service.spec.coffee

```diff
@@ -0,0 +1,288 @@
+        expect(group_result).toEqual
+            name:'Auth'
+            caption: 'Auth related settings'
+            items:[]
+        
+    it 'should merge groups when old group is defined with no items', inject (bbSettingsService) ->
+        localStorage.clear()
+        old_group =
+            name:'System'
+            caption: 'System related settings'
+            items:[]
+        new_group =
+            name:'System'
+            caption: 'System related settings'
+            items:[
+                type:'bool'
+                name:'checkbox_system'
+                default_value: false
+            ,
+                type:'bool'
+                name:'checkbox_system2'
+                default_value: false
+            ]
+        group_result = bbSettingsServiceProviderRef._mergeNewGroup(old_group, new_group)
+        expect(group_result).toEqual
+            name:'System'
+            caption: 'System related settings'
+            items:[
+                type:'bool'
+                name:'checkbox_system',
+                value:false
+                default_value: false
+            ,
+                type:'bool'
+                name:'checkbox_system2',
+                value:false
+                default_value: false
+            ]
+
+    it 'should merge groups when new group is undefined', inject (bbSettingsService) ->
+        localStorage.clear()
+        old_group =
+            name:'System'
+            caption: 'System related settings'
+            items:[
+                type:'bool'
+                name:'checkbox_system'
+                default_value: false
+            ,
+                type:'bool'
+                name:'checkbox_system2'
+                default_value: false
+            ]
+        group_result = bbSettingsServiceProviderRef._mergeNewGroup(old_group, undefined)
+        expect(group_result).toBeUndefined()
+
+    it 'should merge groups when old group is undefined', inject (bbSettingsService) ->
+        localStorage.clear()
+        new_group =
+            name:'Auth'
+            caption: 'Auth related settings'
+            items:[
+                type:'bool'
+                name:'radio1'
+                default_value: false
+            ,
+                type:'bool'
+                name:'radio2'
+                default_value: false
+            ]
+        group_result = bbSettingsServiceProviderRef._mergeNewGroup(undefined, new_group)
+        expect(group_result).toEqual
+            name:'Auth'
+            caption: 'Auth related settings'
+            items:[
+                type:'bool'
+                name:'radio1'
+                value: false
+                default_value: false
+            ,
+                type:'bool'
+                name:'radio2'
+                value: false
+                default_value: false
+            ]
+
+    it 'should merge groups when new group has no name', inject (bbSettingsService) ->
```

Comment:
it 'should not add a group without name ' is maybe a better description ? (merge method is not called)